README.md

```diff
@@ -138,9 +138,9 @@ For more instructions (including OS X) check out https://gist.github.com/wolfv/f
 
 ### Development installation
 
-Make sure to have the following requirements in your conda environment:
+Make sure you install mamba's requirements in your conda environment:
 
-`mamba install cmake compilers pybind11 libsolv libarchive libcurl nlohmann_json pip cpp-filesystem yaml-cpp reproc-cpp -c conda-forge`
+`mamba install --file requirements.txt -c conda-forge`
```

Comment:
It would be more conda-centric to use the command `mamba env update --name mamba-dev --file environment-dev.yml`. Then we can get rid of `requirements.txt`.

Reply:
I think `conda` should be added to `environment-dev.yml` as per the comment in `README.md` about building mamba in a different environment than base.

I have success with the following commands:
```bash
docker run --rm -it condaforge/mambaforge
git clone https://github.com/Aratz/mamba.git
cd mamba
git checkout fix/841

echo "  - conda" >> environment-dev.yml
rm requirements.txt

mamba env update --name mamba-dev --file environment-dev.yml

conda activate mamba-dev
pip install -e .
```

Reply:
Ah, this is what I initially proposed in 6f0baae. What's the main difference between `conda install` and `conda env update` ? They seem quite interchangeable to me so I'd be fine either way.

Reply:
@Aratz, that's a very good question. I wish I knew the answer.

I pulled that particular command from a script I wrote a while back. At the time I think I had a good reason for choosing `conda env update`. It may have been that I was working around some conda bug which has since been fixed. Unfortunately some aspects of conda remind me of [git](https://xkcd.com/1597/) ;)

Reply:
I also find it very bizarre that mamba doesn't install `0.7.18` in the first place.

Perhaps there's some conflict with having all packages in their latest versions, and it pseudorandomly decides that `libsolv` loses?

Reply:
> I think `conda` should be added to `environment-dev.yml` as per the comment in `README.md` about building mamba in a different environment than base.
> 
> I have success with the following commands:
> 
> ```shell
> docker run --rm -it condaforge/mambaforge
> git clone https://github.com/Aratz/mamba.git
> cd mamba
> git checkout fix/841
> 
> echo "  - conda" >> environment-dev.yml
> rm requirements.txt
> 
> mamba env update --name mamba-dev --file environment-dev.yml
> 
> conda activate mamba-dev
> pip install -e .
> ```

you're right, `conda` is missing!

> Ah, this is what I initially proposed in [6f0baae](https://github.com/mamba-org/mamba/commit/6f0baaea48bb2c4384dba93ed760db70327dd9d5). What's the main difference between `conda install` and `conda env update` ? They seem quite interchangeable to me so I'd be fine either way.

We have to distinguish `conda` and `mamba` here.
You point an issue about `mamba` implementation that only handles install-like command using a spec file with `env` sub-command:
https://github.com/mamba-org/mamba/blob/38b374c83952385fc3d7f9fc57e2a264b924bbd2/mamba/mamba_env.py#L107

We should mentioned that in the original issue @Aratz opened.
Maybe it could also help to run use `micromamba` while we are fixing this:
```
docker run --rm -it mambaorg/micromamba
apt update && apt install git -y
git clone https://github.com/mamba-org/mamba.git
cd mamba

echo "  - conda" >> environment-dev.yml
micromamba create -f environment-dev.yml
micromamba activate mamba-dev
pip install -e .
```